app/src/main/java/com/example/randomusergenerator/MainActivity.kt

```diff
@@ -24,16 +36,34 @@ class MainActivity : ComponentActivity() {
                     modifier = Modifier.fillMaxSize(),
                     color = MaterialTheme.colorScheme.background
                 ) {
-                    UserScreen()
+                    MyApp()
                 }
             }
         }
     }
-}
-@Preview(showBackground = true)
-@Composable
-fun DefaultPreview() {
-    RandomUserGeneratorTheme {
-        UserScreen()
+
+    @Composable
+    fun MyApp() {
+        val navController = rememberNavController()
+        val userViewModel: UserViewModel = hiltViewModel()
```

Comment:
Hmmmmm
I don't think you are meant to call the `UserViewModel` in the activity. What if there were several `ViewModel` classes? That would make all of their lifecycles attached to the activity which should not be rather to the view observing it.

So, this function needs to be refactored to not observe the viewmodel except the navigation. Alos, there should be a workaround to the user details.

Reply:
its not necessarily in the activity, its in another composable. I can extract the composable to another class

Reply:
You should not reference the viewmodel outside the main caller/user. You might need to create a navigator that would do that for you. Then, navigate from your viewmodel instead from the view.